backup.nix

```diff
@@ -0,0 +1,133 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  secretsDir = config.nix-bitcoin.secretsDir;
+
+  # Use borg 1.2.1 (the latest 1.2.* release)
+  # TODO-EXTERNAL: Remove this when nixpkgs-unstable has been updated
+  nixpkgs_borg_1_2_1 = pkgs.fetchFromGitHub {
+    owner = "NixOS";
+    repo = "nixpkgs";
+    rev = "6616de389ed55fba6eeba60377fc04732d5a207c";
+    sha256 = "1h8lvyrv4sb5fhimzniiw6zjn74hl30zm9g8nzcaq331bd20gpw6";
+  };
+  pkgs_borg_1_2_1 = import nixpkgs_borg_1_2_1 {
+    inherit (pkgs.stdenv) system;
+    config = {};
+    overlays = [];
+  };
+in
+{
+  services.zfs.autoSnapshot.enable = true;
+
+  # Only use daily, weekly, monthly ZFS snapshots
+  systemd.timers = {
+    zfs-snapshot-frequent.enable = false;
+    zfs-snapshot-hourly.enable = false;
+    # The daily snapshot is run by borgbackup-job-main.service
+    zfs-snapshot-daily.enable = false;
+  };
+
+  systemd.services.borgbackup-job-main = rec {
+    requires = [ "zfs-snapshot-daily.service" ];
+    after = requires;
+  };
+
+  programs.ssh.knownHosts."freak.seedhost.eu".publicKey = "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBD2TmdE89ZD4XshcIcXZPLFC/nDxZdAr9yrH2/2OCNKEo/Ex60y8TQjp93isjdDj7Grf/GpW60OONfXTFe0r5iM=";
+
+  services.borgbackup.jobs = {
+    main = {
+      startAt = "daily";
+
+      preHook = ''
+        latest_daily_snap=$(
+          shopt -s nullglob
+          printf '%s\n' /.zfs/snapshot/*daily* | tail -1
+        )
+        if [[ ! $latest_daily_snap ]]; then
+          echo "Error: No daily snapshot found"
+          exit 1
+        fi
+        echo "Using $latest_daily_snap"
+        cd $latest_daily_snap
+      '';
+
+      paths = [ "var/lib" ];
+
+      exclude = [
+        "var/lib/bitcoind/blocks"
+        "var/lib/bitcoind/chainstate"
+        "var/lib/bitcoind/indexes"
+        "var/lib/liquidd/*/blocks"
+        "var/lib/liquidd/*/chainstate"
+        "var/lib/liquidd/*/indexes"
+        "var/lib/electrs"
+        "var/lib/fulcrum"
+        "var/lib/nbxplorer"
+        "var/lib/duplicity"
+        "var/lib/onion-addresses"
+
+        "var/lib/i2pd"
+        "var/lib/redis"
+        "var/lib/udisks2"
+        "var/lib/usbguard"
+
+        "var/lib/acme"
+        "var/lib/dovecot"
+        "var/lib/dhparams" # from dovecot
+        "var/lib/postfix"
+        "var/lib/rspamd"
+
+        "var/lib/machines"
+        "var/lib/private"
+        "var/lib/systemd"
+      ];
+
+      repo = "nixbitcoin@freak.seedhost.eu:borg-backup";
+      doInit = false;
+      encryption = {
+        mode = "repokey";
+        passCommand = "cat ${secretsDir}/backup-encryption-password";
```

Comment:
I think we can safely delete `${secretsDir}/backup-encryption-env` now that duplicity is no longer in use.